reference/auth.v1.yaml

```diff
@@ -680,6 +680,150 @@ paths:
         - sessionToken: []
       tags:
         - Internal
+  /v1/consents:
+    get:
+      summary: List Consents
+      tags: []
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: ./auth/models/consent/consents.v1.yaml
+      operationId: ListConsents
+      description: Retrieves the list of all different consent types and their content. By default returns only the latest version for each consent type. This can be overriden by setting the `version` query parameter to `all`
+      requestBody:
+        content: {}
+      parameters:
+        - schema:
+            type: string
+            enum:
+              - latest
+              - all
+            default: latest
+          in: query
+          name: version
+          description: Whether to return all versions or only the most recent one.
+    parameters: []
+  '/v1/consents/{consentType}':
+    parameters:
+      - $ref: '#/components/parameters/consentType'
+    get:
+      summary: Get Latest Consent By Type
+      tags: []
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: ./auth/models/consent/consent.v1.yaml
+      operationId: GetLatestConsentByType
+      description: Returns the latest consent version for the given type
+  '/v1/consent/{consentType}/versions':
+    parameters:
+      - $ref: '#/components/parameters/consentType'
+    get:
+      summary: Get Consent Versions
+      tags: []
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: ./auth/models/consent/consents.v1.yaml
+      operationId: GetConsentVersions
+      description: Returns a list of all consent versions for a given type
+  '/v1/users/{userId}/consents':
+    parameters:
+      - $ref: '#/components/parameters/userId'
+    get:
+      summary: Get User Consent Records
+      tags: []
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: ./auth/models/consent/records.v1.yaml
+      operationId: GetUserConsentRecords
+      description: Returns a list of all user consent records. By default only the most recent consent record for a given consent type will be returned. This can be overriden by setting the `version` query parameter to `all`.
+      parameters:
+        - schema:
+            type: string
+            enum:
```

Comment:
What other values would ever be added to this enum? Would this be better as a boolean flag? That is, `?allVersions=true`